src/js/actions/guides.js

```diff
@@ -196,47 +200,101 @@ define(function (require, exports) {
     createGuideAndTrack.writes = [locks.JS_DOC];
 
     /**
-     * Updates the given guide's position or creates a new guide if necessary
+     * Helper function to figure out whether the guide with the
+     * given position and orientation is within the visible canvas bounds
+     *
+     * @private
+     * @param {string} orientation "horizontal" or "vertical"
+     * @param {number} position
+     * @return {boolean} True if guide is within the canvas bounds
+     */
+    var _guideWithinVisibleCanvas = function (orientation, position) {
+        var uiStore = this.flux.store("ui"),
+            cloakRect = uiStore.getCloakRect(),
+            horizontal = orientation === "horizontal",
+            x = horizontal ? 0 : position,
+            y = horizontal ? position : 0,
+            windowPos = uiStore.transformCanvasToWindow(x, y);
+
+        if (horizontal &&
+            (windowPos.y < cloakRect.top || windowPos.y > cloakRect.bottom)) {
+            return false;
+        } else if (!horizontal &&
+            (windowPos.x < cloakRect.left || windowPos.x > cloakRect.right)) {
+            return false;
+        } else {
+            return true;
+        }
+    };
+
+    /**
+     * Deletes the given guide
      *
      * @param {object} payload
      * @param {number} payload.documentID Owner document ID
      * @param {number} payload.index Index of the edited guide
-     * @param {string} payload.orientation New orientation of the guide
-     * @param {number} payload.position New position of the guide
+     * @param {boolean=} sendChanges If true, will call the action descriptor to delete the guide from PS
      *
      * @return {Promise}
      */
-    var setGuide = function (payload) {
-        return this.dispatchAsync(events.document.history.nonOptimistic.GUIDE_SET, payload)
+    var deleteGuide = function (payload, sendChanges) {
```

Comment:
It started with the fact that initially I was immediately dispatching the information to the stores, so I didn't feel the need to unwrap things and wrap them back again. But now that we're calling adapter functions, I should make the change.